Feature/addOnPasteHook

sashless opened the pull request:
for https://github.com/stevermeister/ngWig/issues/59

sashless commented:
well it works basically but the buttons stopped working for me. Dont know if that is related, investigating

stevermeister commented:
@sashless 
"but the buttons stopped working for me" - but you can check it returning back to master, right?

sashless commented:
Yea right, will figure that out tomorrow

Stepan Suvorov <notifications@github.com> schrieb am Di., 29. Sep. 2015
18:04:

> @sashless <https://github.com/sashless>
> "but the buttons stopped working for me" - but you can check it returning
> back to master, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevermeister/ngWig/pull/81#issuecomment-144104307>.
>

sashless commented:
okay i fixed that @stevermeister 

i have removed the version bump because i dont have any push permissions anyway =)

Please release afterwards =)

sashless commented:
I have updated the demo page as well if you want to try the new behavior =)

sashless commented:
actually i merged in master and there were a lot of changes, need to test it again

sashless commented:
allright @stevermeister still works :+1:

sashless commented on src/javascript/app/ng-wig/ngWigEditableDirective.js:
```diff
@@ -14,7 +14,15 @@ angular.module('ngWig')
-      $element.bind('blur keyup change paste focus click', function() {
+      if (angular.isFunction(scope.onPaste)) {
+        $element.on('paste', function(e) {
+          scope.onPaste(e).then(function(val) {
```
actually it would be better to provide the event AND the current content.

This way we can not just replace the current content. We can ADD additional paste content as well

sashless replied:
will do this later in the week

sashless commented:
this is done now @stevermeister

stevermeister commented on README.md:
```diff
@@ -21,3 +21,8 @@ More information on [ngWig site](http://stevermeister.github.io/ngWig/)
 
 [![Screenshot] (http://stevermeister.github.io/ngWig/images/ng-wig-demo.png)](http://stevermeister.github.io/ngWig/demo/)
+
+## Development Setup
```
:+1:

stevermeister commented on package.json:
```diff
@@ -2,6 +2,9 @@
   "version": "2.1.0",
   "author": "Stepan Suvorov",
+  "contributors": [
+    {"name": "Sascha Weniger", "url": "https://github.com/sashless"}
```
:)

could you please then put all the names(http://stepan.js.org/ngWig/)? 
plus - https://github.com/bolelamx

sashless replied:
sure :+1:

sashless replied:
done :+1: @stevermeister

besuhoff commented:
@sashless what about other event hooks? onCommandExec, onChange, etc? For me event callback strategy looks more like a piece of a common behavior

sashless commented:
sorry but i am not sure what are you asking for @besuhoff ?

sashless commented:
can we go ahead here @stevermeister ?

stevermeister commented on src/javascript/app/ng-wig/ngWigEditableDirective.js:
```diff
@@ -14,7 +14,25 @@ angular.module('ngWig')
+
+      if (angular.isFunction(scope.onPaste)) {
+        $element.on('paste', function(e) {
+          scope.onPaste(e, $element.html()).then(function(val) {
```
are you sure that it should be async operation?

sashless replied:
Yea I am sure. Usually this usecase should not require to be async but in
case someone needs it for $timeout for example it will work as well.

Stepan Suvorov <notifications@github.com> schrieb am So., 11. Okt. 2015
16:05:

> In src/javascript/app/ng-wig/ngWigEditableDirective.js
> <https://github.com/stevermeister/ngWig/pull/81#discussion_r41710786>:
>
> > @@ -14,7 +14,25 @@ angular.module('ngWig')
> >          ngModelController.$setViewValue($element.html());
> >        }
> >
> > -      $element.bind('blur keyup change paste focus click', function() {
> > +      var eventsToBind = [
> > +        'blur',
> > +        'keyup',
> > +        'change',
> > +        'focus',
> > +        'click'
> > +      ];
> > +
> > +      if (angular.isFunction(scope.onPaste)) {
> > +        $element.on('paste', function(e) {
> > +          scope.onPaste(e, $element.html()).then(function(val) {
>
> are you sure that it should be async operation?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevermeister/ngWig/pull/81/files#r41710786>.
>

stevermeister commented on src/javascript/app/ng-wig/ngWigEditableDirective.js:
```diff
@@ -14,7 +14,25 @@ angular.module('ngWig')
+        eventsToBind.push('paste');
+      }
+
+      $element.bind(eventsToBind.join(' '), function() {
```
you can put 'paste' event inside. 
it should make code more readable, 
what do you think?

sashless replied:
When put inside intercepting the paste event and altering content does not
work.

Stepan Suvorov <notifications@github.com> schrieb am So., 11. Okt. 2015
16:06:

> In src/javascript/app/ng-wig/ngWigEditableDirective.js
> <https://github.com/stevermeister/ngWig/pull/81#discussion_r41710805>:
>
> > +        'change',
> > +        'focus',
> > +        'click'
> > +      ];
> > +
> > +      if (angular.isFunction(scope.onPaste)) {
> > +        $element.on('paste', function(e) {
> > +          scope.onPaste(e, $element.html()).then(
>
> function(val) {
> > +            $element.html(val);
> > +          })
> > +        });
> > +      }else{
> > +        eventsToBind.push('paste');
> > +      }
> > +
> > +      $element.bind(eventsToBind.join(' '), function() {
>
> you can put 'paste' event inside.
> it should make code more readable,
> what do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevermeister/ngWig/pull/81/files#r41710805>.
>

stevermeister commented:
ok, let's try

sashless commented:
please release a new version and publish to NPM @stevermeister including this =)

stevermeister commented:
done, v2.1.1 is there, enjoy!

sashless commented:
ah well, this is more a new feature than a bug fix. According to SemVer `2.2.0` would be appropriate.

```
 MAJOR version when you make incompatible API changes,
MINOR version when you add functionality in a backwards-compatible manner, and
PATCH version when you make backwards-compatible bug fixes.
```

 thanks!

stevermeister commented:
@sashless 
makes sense, updated

sashless commented:
ay thanks ! Actually i missed to update the README providing informations about the `onPaste` callback. I will file a PR soon !
